server/src-lib/Hasura/RQL/DDL/EventTrigger.hs

```diff
@@ -307,6 +309,24 @@ runInvokeEventTrigger (InvokeEventTriggerQuery name payload) = do
       Just True -> return ()
       _         -> throw400 NotSupported "manual mode is not enabled for event trigger"
 
+runPauseEventTrigger
+  :: (MonadTx m)
+  => PauseEventTriggerQuery -> m EncJSON
+runPauseEventTrigger (PauseEventTriggerQuery name) = do
+  affectedRows <- updatePauseOfEventTriggerInCatalog name True
+  if | affectedRows == 1 -> pure successMsg
+     | affectedRows == 0 -> throw400 NotFound "event trigger not found"
+     | otherwise -> throw500 "unexpected: more than one event triggers have been paused"
+
+runResumeEventTrigger
+  :: (MonadTx m)
+  => ResumeEventTriggerQuery -> m EncJSON
+runResumeEventTrigger (ResumeEventTriggerQuery name) = do
+  affectedRows <- updatePauseOfEventTriggerInCatalog name False
+  if | affectedRows == 1 -> pure successMsg
+     | affectedRows == 0 -> throw400 NotFound "event trigger not found"
+     | otherwise -> throw500 "unexpected: more than one event triggers have been resumed"
```

Comment:
This logic is duplicated between `runPauseEventTrigger` and `runResumeEventTrigger`, but I don’t think it needs to be—this is really an implementation detail of `updatePauseOfEventTriggerInCatalog`! So I think you could move this check into that function, and that way it wouldn’t need to be duplicated at both call sites.